src/Controller/Link/Create.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller\Link;
+
+use App\Entity\Link;
+use App\Repository\LinkRepositoryInterface;
+use App\Service\RandomString\GeneratorInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class Create
+{
+    private LinkRepositoryInterface $repository;
+    private GeneratorInterface $generator;
+
+    public function __construct(LinkRepositoryInterface $repository, GeneratorInterface $generator)
+    {
+        $this->repository = $repository;
+        $this->generator = $generator;
+    }
+
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response): ResponseInterface
+    {
+        $data = $request->getParsedBody();
+
+        $links = [];
+        foreach ($data as $linkData) {
+            $this->repository->save($links[] = new Link(
+                $this->generator->getRandomString(),
+                ...array_values(
+                    array_merge(
+                        [
+                            'long_url' => '',
+                            'title' => null,
+                            'tags' => [],
+                        ],
+                        $linkData
+                    )
+                )
+            ));
```

Comment:
Раз решил использовать минимальное количество пакетов, и реализовывать свой вариант DTO,
то пожалуй это все должно выглядеть, где-то так, по причине улучшения читаемости и использования возможностей языка:
```suggestion
$links[$key] = new Link($this->generator->getRandomString(), ...$linkData);
$this->repository->save($links[$key]);
```
Я немного переработал что-бы все отрабатывало, но думаю основную идею ты понял
http://prntscr.com/26p6qdl
Но в итоге DTO должно выглядеть как-то так
http://prntscr.com/26p6onw